File: algorithms/code/leetcode/lc028_implement_str_str/LC028ImplementStrStr.java
package leetcode.lc028_implement_str_str;

/**
 * https://leetcode.com/problems/implement-strstr/
 */
public final class LC028ImplementStrStr {
    public int strStr(final String haystack, final String needle) {
        if (haystack.length() < needle.length()) {
            return -1;
        }
        for (int i = 0; i <= haystack.length() - needle.length(); i++) {
            if (haystack.substring(i, i + needle.length()).equals(needle)) {
                return i;
            }
        }
        return -1;
    }
}

File: algorithms/code/leetcode/lc028_implement_str_str/lc028_implement_str_str.go
package lc028

// https://leetcode.com/problems/implement-strstr/

// StrStr function implements the substring search algorithm.
func StrStr(haystack string, needle string) int {
	if len(haystack) < len(needle) {
		return -1
	}
	for i := 0; i <= len(haystack)-len(needle); i++ {
		if haystack[i:i+len(needle)] == needle {
			return i
		}
	}
	return -1
}
